pallets/acurast-fulfillment-receiver/src/mock.rs

```diff
@@ -0,0 +1,138 @@
+    }
+);
+
+parameter_types! {
+    pub const BlockHashCount: BlockNumber = 2400;
+}
+parameter_types! {
+    pub BlockWeights: frame_system::limits::BlockWeights = frame_system::limits::BlockWeights::simple_max(Weight::from_ref_time(1024));
+    pub const MinimumPeriod: u64 = 6000;
+    pub AllowedFulfillAccounts: Vec<AccountId> = vec![bob_account_id()];
+}
+parameter_types! {
+    pub const MaxReserves: u32 = 50;
+    pub const MaxLocks: u32 = 50;
+}
+parameter_types! {
+    pub const AcurastPalletId: PalletId = PalletId(*b"acrstpid");
+}
+
+impl frame_system::Config for Test {
+    type Call = Call;
+    type Index = u32;
+    type BlockNumber = BlockNumber;
+    type Hash = sp_core::H256;
+    type Hashing = BlakeTwo256;
+    type AccountId = AccountId;
+    type Lookup = AccountIdLookup<AccountId, ()>;
+    type Header = generic::Header<BlockNumber, BlakeTwo256>;
+    type Event = Event;
+    type Origin = Origin;
+    type BlockHashCount = BlockHashCount;
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = ();
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type DbWeight = ();
+    type BaseCallFilter = Everything;
+    type SystemWeightInfo = ();
+    type BlockWeights = ();
+    type BlockLength = ();
+    type SS58Prefix = ();
+    type OnSetCode = ();
+    type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+impl crate::Config for Test {
+    type Event = Event;
+    type OnFulfillment = FulfillmentHandler;
+    type WeightInfo = ();
+}
+
+pub struct FulfillmentHandler;
+impl OnFulfillment<Test> for FulfillmentHandler {
+    fn on_fulfillment(
+        from: <Test as frame_system::Config>::AccountId,
+        _fulfillment: crate::Fulfillment,
+    ) -> sp_runtime::DispatchResultWithInfo<frame_support::weights::PostDispatchInfo> {
+        if !AllowedFulfillAccounts::get().contains(&from) {
+            return Err(DispatchError::BadOrigin.into());
+        }
+        Ok(().into())
+    }
+}
+
+pub fn alice_account_id() -> AccountId {
+    [0; 32].into()
+}
+
+pub fn bob_account_id() -> AccountId {
+    [1; 32].into()
+}
+
+pub struct ExtBuilder;
+
+impl ExtBuilder {
+    pub fn build(self) -> sp_io::TestExternalities {
+        let t = frame_system::GenesisConfig::default()
+            .build_storage::<Test>()
+            .unwrap();
+
+        let mut ext = sp_io::TestExternalities::new(t);
+        ext.execute_with(|| System::set_block_number(1));
+        ext
+    }
+}
+
+impl Default for ExtBuilder {
+    fn default() -> Self {
+        Self {}
+    }
+}
+
+pub const SCRIPT_BYTES: [u8; 53] = hex!("697066733A2F2F00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");
+
+pub fn script() -> Script {
+    SCRIPT_BYTES.to_vec().try_into().unwrap()
+}
+
+pub fn fulfillment_for(script: Script) -> Fulfillment {
+    Fulfillment {
+        script,
+        payload: hex!("00").to_vec(),
+    }
+}
```

Comment:
Non-blocking: These mocks cannot be used for benchmarking when the pallet's user performs his duty to write the benchmarks to generate `WeightInfo`, because some types like `frame_system::GenesisConfig` are not available in benchmark builds (== with `runtime-benchmark` feature without `test` flag). 
I started using this module structure in `lib.rs` to solve this:

```
#[cfg(test)]
mod mock;
#[cfg(any(test, feature = "runtime-benchmarks"))]
pub mod stub;
```